plugin/trino-ignite/src/main/java/io/trino/plugin/ignite/ImplementAvgDecimal.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.ignite;
+
+import io.trino.matching.Capture;
+import io.trino.matching.Captures;
+import io.trino.matching.Pattern;
+import io.trino.plugin.jdbc.JdbcColumnHandle;
+import io.trino.plugin.jdbc.JdbcExpression;
+import io.trino.plugin.jdbc.expression.AggregateFunctionRule;
+import io.trino.spi.connector.AggregateFunction;
+import io.trino.spi.expression.Variable;
+import io.trino.spi.type.DecimalType;
+
+import java.util.Optional;
+
+import static com.google.common.base.Verify.verify;
+import static io.trino.matching.Capture.newCapture;
+import static io.trino.plugin.jdbc.expression.AggregateFunctionPatterns.basicAggregation;
+import static io.trino.plugin.jdbc.expression.AggregateFunctionPatterns.expressionType;
+import static io.trino.plugin.jdbc.expression.AggregateFunctionPatterns.functionName;
+import static io.trino.plugin.jdbc.expression.AggregateFunctionPatterns.singleInput;
+import static io.trino.plugin.jdbc.expression.AggregateFunctionPatterns.variable;
+import static java.lang.String.format;
+
+/**
+ * Implements {@code avg(decimal(p, s)}
+ */
+public class ImplementAvgDecimal
+        implements AggregateFunctionRule
+{
+    private static final Capture<Variable> INPUT = newCapture();
+
+    @Override
+    public Pattern<AggregateFunction> getPattern()
+    {
+        return basicAggregation()
+                .with(functionName().equalTo("avg"))
+                .with(singleInput().matching(
+                        variable()
+                                .with(expressionType().matching(DecimalType.class::isInstance))
+                                .capturedAs(INPUT)));
+    }
+
+    @Override
+    public Optional<JdbcExpression> rewrite(AggregateFunction aggregateFunction, Captures captures, RewriteContext context)
+    {
+        Variable input = captures.get(INPUT);
+        JdbcColumnHandle columnHandle = (JdbcColumnHandle) context.getAssignment(input.getName());
+
+        DecimalType type = (DecimalType) columnHandle.getColumnType();
+        verify(aggregateFunction.getOutputType().equals(type));
+
+        String column = context.getIdentifierQuote().apply(columnHandle.getColumnName());
+
+        // wait https://issues.apache.org/jira/browse/IGNITE-14948 to be solved.
+        return Optional.of(new JdbcExpression(
```

Comment:
I need to look at this deeper but from a quick look there can be issues when `precision = scale`.

`decimal(p, s)` gets returned as `decimal(p+1, s)` when it should be `decimal(p+1,s+1)`?

Reply:
Can we add more test cases for the scenario?

Reply:
Yes we should. Something like avg of two `DECIMAL(38, 10)` values so that doing `p+1` becomes 39.
And another with `DECIMAL(38,38)` values so that both `p+1` and `s+1` become 39.

Maybe this should be added to BaseJdbcConnectorTest as `testAvgDecimalExceedingSupportedPrecision` (or a better name)?

cc: @findepi @wendigo

Reply:
Should create a new pr to add this test case before this?

Reply:
For now let's add the test in Ignite connector only. As a follow-up we can extract the common parts to some base class.